### PR TITLE
refactor: datetime time unit

### PR DESCRIPTION
**Pull request by QuenKar**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

_PLEASE DO NOT LEAVE THIS EMPTY !!!_

- change datetime's default time unit from second to millisecond

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/common/time/src/datetime.rs [hunk 1]
```diff
@@ -84,6 +85,13 @@ impl From<i64> for DateTime {
     }
 }
 
+impl From<Date> for DateTime {
+    fn from(value: Date) -> Self {
+        // It's safe, i32 * 86400000 won't be overflow
+        Self(value.to_secs() * 1000)
```
Comment (evenyag) [inline src/common/time/src/datetime.rs:91]: Do we have a test for this?

File: src/common/time/src/datetime.rs [hunk 2]
```diff
@@ -96,6 +104,10 @@ impl DateTime {
     pub fn to_chrono_datetime(&self) -> Option<NaiveDateTime> {
         NaiveDateTime::from_timestamp_opt(self.0, 0)
     }
+
+    pub fn date(&self) -> Option<Date> {
```
Comment (evenyag) [inline src/common/time/src/datetime.rs:108]:
```suggestion
    pub fn to_date(&self) -> Option<Date> {
```

File: src/common/time/src/datetime.rs (hunk 1 above)
Comment (QuenKar) [reply src/common/time/src/datetime.rs:91]: I will add a test.

**Review by evenyag — APPROVED** [review]

**Comment by v0y4g3r** [comment]
Maybe add some sqlness test to ensure this behavior?

**Review by v0y4g3r — APPROVED** [review]
LGTM

File: src/common/time/src/datetime.rs [hunk 3]
```diff
@@ -96,6 +104,10 @@ impl DateTime {
     pub fn to_chrono_datetime(&self) -> Option<NaiveDateTime> {
         NaiveDateTime::from_timestamp_opt(self.0, 0)
```
Comment (zhongzc) [inline src/common/time/src/datetime.rs:105]: the first param for `from_timestamp_opt` is secs

**Pull request merged**